client/src/components/ElementEditor/ElementList.js

```diff
@@ -39,6 +39,13 @@ class ElementList extends Component {
       this.resetState(prevState, false);
       return;
     }
+    // Scenario we've just clicked "save" or "submit" on the form
+    if (this.state.saveAllElements && !prevState.saveAllElements) {
+      // Reset the validation state of all blocks.
+      // This mirrors handleBeforeSubmitForm() for individual blocks.
+      this.resetState(prevState, false);
+      return;
+    }
```

Comment:
Needed because otherwise we retain the old "invalid" state. As noted in the code comment, resetting the validation state is already done when saving a block with the `...` menu, so makes sense to reset it when submitting via the page save button as well.